### PR TITLE
[CHEC-980] Table component w/ prepend

### PR DESCRIPTION
**Pull request by kvisca**
Adding table component so we can have prepends

Screen Shot 2020-10-22 at 5 08 29 PM

### CODE REVIEW

File: src/components/ChecTable.vue [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  <div class="chec-table" :class="classNames">
+    <div v-if="$slots.prepend" class="chec-table__prepend">
+      <div class="chec-table__inner-wrapper">
+        <!--
+          @slot Search, tags etc.
+        -->
+        <slot name="prepend" />
+      </div>
+    </div>
+    <table class="table" :class="tableClass">
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:11]:
```suggestion
    <table class="chec-table__table" :class="tableClass">
```

File: src/components/ChecTable.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<script>
+export default {
+  name: 'ChecTable',
+  props: {
+    /**
+     * Class to pass to table
+     */
+    tableClass: {
+      type: String,
+      default: 'table-auto',
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:29]:
```suggestion
      default: '',
```

File: src/components/ChecTable.vue [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.chec-table {
+  &__prepend {
+    @apply relative px-4 rounded-t-md bg-gray-200;
+    z-index: 1;
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:48]: Do you still need this z-index?

File: src/components/ChecTable.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  &__inner-wrapper {
+    @apply bg-gray-200 py-4 border-b border-solid border-gray-300;
+  }
+
+  .table {
+    @apply relative;
+    z-index: 2;
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:57]: And this z-index?

File: src/components/ChecTable.vue [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  &__prepend {
+    @apply relative px-4 rounded-t-md bg-gray-200;
+    z-index: 1;
+  }
+
+  &__inner-wrapper {
+    @apply bg-gray-200 py-4 border-b border-solid border-gray-300;
+  }
+
+  .table {
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:55]:
```suggestion
  &__table {
```

File: src/components/ChecTable.vue [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    @apply relative;
+    z-index: 2;
+  }
+
+  &--prepended {
+    @apply shadow rounded;
+
+    .table {
```
Comment (ScopeyNZ) [inline src/components/ChecTable.vue:63]:
```suggestion
    .chec-table__table {
```

**Review by ScopeyNZ — COMMENTED** [review]
Looking good!

File: src/components/ChecTable.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+</template>
+
+<script>
+export default {
+  name: 'ChecTable',
+  props: {
+    /**
+     * Class to pass to table
+     */
+    tableClass: {
```
Comment (robbieaverill) [inline src/components/ChecTable.vue:27]: Can you call this `innerClass`? We use that in ChecCard already, and we use `innerInputClass` in Text and Password field, it'd be good to standardise on something and I think `innerClass` is a good prop name for that

**Review by ScopeyNZ — APPROVED** [review]

**Pull request merged**